### PR TITLE
Detect union ctype

### PR DESCRIPTION
**Pull request by EPashkin**
For https://github.com/gtk-rs/gtk/pull/340

### CODE REVIEW

File: src/library_postprocessing.rs [hunk 1]
```diff
@@ -130,6 +130,17 @@ impl Library {
         if glib_name.is_none() { return None; }
         let glib_name = glib_name.unwrap();
         let detected_c_type = match *type_ {
+            Alias(ref alias) => {
+                if let Some(aliased) = self.c_type_by_type_id(alias.typ) {
+                    if aliased.find("*").is_some() {
+                        format!("{}*", glib_name)
```
Comment (gkoz) [inline src/library_postprocessing.rs]: What's going to happen if it's something different from `TypeName*` (perhaps a const pointer)?

Comment (EPashkin) [inline src/library_postprocessing.rs]: It called recursively and I don't see any other cases as other data used.

Comment (EPashkin) [inline src/library_postprocessing.rs]: * as no other data used

Comment (EPashkin) [inline src/library_postprocessing.rs]: I can extract from this function `fn is_pointer(...)->bool` to make logic clearer

**Comment by EPashkin** [comment]
Simplified logic

File: src/library_postprocessing.rs [hunk 2]
```diff
@@ -124,18 +124,26 @@ impl Library {
     }
 
     fn c_type_by_type_id(&self, tid: TypeId) -> Option<String> {
-        use library::Type::*;
         let type_ = self.type_(tid);
         let glib_name = type_.get_glib_name();
```
Comment (gkoz) [inline src/library_postprocessing.rs]: Now I understand. Also after splitting out `is_referenced_type` this bit can become simpler:
```rust
type_.get_glib_name().map(|glib_name| {
...
```

Comment (EPashkin) [inline src/library_postprocessing.rs]: You right
Fixed

**Comment by gkoz** [comment]
Thanks.

**Pull request merged**